src/crystal/system/unix/user.cr

```diff
@@ -4,7 +4,7 @@ module Crystal::System::User
   private GETPW_R_SIZE_MAX = 1024 * 16
 
   private def from_struct(pwd)
-    user = String.new(pwd.pw_gecos).split(",").first
+    user = String.new(pwd.pw_gecos).partition(',')[0]
```

Comment:
Why?

Reply:
`partition` might be a bit faster (it returns a tuple instead of an array)

Reply:
If so, maybe this would read a bit better?

```suggestion
    user = String.new(pwd.pw_gecos).partition(',').first
```

Reply:
`partition` still allocates two unnecessary strings. A more efficient implementation would just scan for the first index of `,` and take a slice. This is even possible before creating a `String` instance, reducing the total amount of memory allocations to one.
```cr
user = String.new(pwd.pw_gecos, pwd.pw_gecos.to_slice(LibC.strlen(pwd.pw_gecos)).index(','.ord))
```

This could even avoid `LibC.strlen` because the `,` index will always be smaller. But this would mean to search through the char pointer manually, making the code more convoluted.

Reply:
Hum, not sure it's worth to complicate that much for the gain, at this point.